### PR TITLE
Upload the file fix 

### PR DESCRIPTION
**Pull request by heron2014**
## Ready to review
Related to [552](https://github.com/FAC-GM/app/issues/552)

@nelsonic please have a look at this small PR, thanks

### CODE REVIEW

File: lib/handlers/csv_list/helpers/add_list_to_candidates.js [hunk 1]
```diff
@@ -32,12 +32,10 @@ module.exports = function( csvFile, listName, next) {
     }
 
     let emails = csvJson.map(csvRow => {
-      return csvRow.Email
+      return csvRow.Email;
     });
 
     //filter the empty emails
-    emails = emails.filter(Boolean);
```
Comment (SimonLab) [inline lib/handlers/csv_list/helpers/add_list_to_candidates.js]: Why don't we keep this filter?

File: lib/handlers/csv_list/helpers/add_list_to_candidates.js [hunk 2]
```diff
@@ -59,10 +56,14 @@ module.exports = function( csvFile, listName, next) {
         /**
         * Filter the candidates where the listName is not already define
         */
-        const candidateUpdates = candidates.filter(candidate => {
 
-          return candidate.listNames.indexOf(listName) === -1
-        })
+        const candidateUpdates = candidates.filter(candidate => {
```
Comment (SimonLab) [inline lib/handlers/csv_list/helpers/add_list_to_candidates.js]: candidateUpdates is the list of candidates who don't belong to the list. So the filter also need to keep the candidates where listNames is not define

```
        const candidateUpdates = candidates.filter(candidate => {
          // $lab:coverage:off$
            if (candidate.listNames) {
              return candidate.listNames.indexOf(listName) === -1;
            } else {
             // the candidate doesn't have the listNames property so she doesn't belong to the list so we need to add her
            return true;
           }
              // $lab:coverage:on$
        });

**Comment by SimonLab** [comment]
On the file database-helpers/elasticsearch/csv_list/update_candidate_listNames.js you also need to add a condition in case the listNames property is not yet defined:

```
  profiles.forEach(profile => {

    let listNamesUpdate = profile.listNames || [];
    listNamesUpdate.push(listName);
    bulk.push({update: {_index: process.env.ES_INDEX, _type: process.env.ES_TYPE, _id: profile.id }})
.
.
.
```

File: lib/handlers/csv_list/helpers/add_list_to_candidates.js (hunk 1 above)
Comment (heron2014) [inline lib/handlers/csv_list/helpers/add_list_to_candidates.js]: Yes this should be there, Thanks

**Comment by heron2014** [comment]
@SimonLab thank you so much!

**Comment by heron2014** [comment]
@SimonLab we defined the listNames property on all the profiles https://github.com/FAC-GM/app/blob/master/lib/helpers/complete_profile.js#L26 , why in database-helpers/elasticsearch/csv_list/update_candidate_listNames.js  we still need the condition to empty array?

**Comment by SimonLab** [comment]
Just in case if there was a problem with Elasticsearch

**Comment by nelsonic** [comment]
LGTM

**Pull request merged**